### ISSUE TITLE
Default Application Filters

### ISSUE THREAD

**Issue by dyatlov** [argoproj/argo-cd#15263]
# Summary

We'd like to have default application filters in UI to limit what applications are visible to a user.

# Motivation

TLDR: we have some "garbage" applications consisting of just a single resource and the ratio of good / garbage app is 1:10.

In order for some of our applications to work we need to make resource changes in other clusters and namespaces upon deployment of an application.
These "resource changes" are deployed as other applications.
That results in a lot of automated apps which makes no real use for us but ArgoCD UI still loads all of them and each time we have to filter out them manually.

# Proposal

Add default filters into `argocd-cm.yaml` and then the UI should pass on these filters to backend when loading application list.
The filters can be visible in sidebar as normal ones. 
For example, if we added a default filter by label, then upon loading ArgoCD UI that label should be already preselected in sidebar.

**Comment by saumeya** [comment]
Hi @dyatlov how about using `favorite filter` for the necessary applications?
